backend/infrahub/graphql/types/event.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from graphene import Field, Interface, ObjectType, String
+from graphene.types.generic import GenericScalar
+
+if TYPE_CHECKING:
+    from graphql import GraphQLResolveInfo
+
+
+class EventNodeInterface(Interface):
+    id = String(required=True)
+    event = String(required=True)
+    branch = String(required=False)
+
+    @classmethod
+    def resolve_type(cls, instance: dict[str, Any], info: GraphQLResolveInfo) -> ObjectType:
+        if "event" in instance:
+            return EVENT_TYPES.get(instance["event"], StandardEvent)
+        return StandardEvent
+
+
+class EventNodes(ObjectType):
+    node = Field(EventNodeInterface)
+
+
+# ---------------------------------------
+# Branch events
+# ---------------------------------------
+class BranchCreateEvent(ObjectType):
+    class Meta:
+        interfaces = (EventNodeInterface,)
+
+    payload = Field(GenericScalar, required=True)
+
+
+class BranchRebaseEvent(ObjectType):
+    class Meta:
+        interfaces = (EventNodeInterface,)
+
+    payload = Field(GenericScalar, required=True)
+
+
+class BranchDeleteEvent(ObjectType):
+    class Meta:
+        interfaces = (EventNodeInterface,)
+
+    payload = Field(GenericScalar, required=True)
+
+
+# ---------------------------------------
+# Node/Object events
+# ---------------------------------------
+class NodeAddedEvent(ObjectType):
+    class Meta:
+        interfaces = (EventNodeInterface,)
+
+    payload = Field(GenericScalar, required=True)
+
+
+class StandardEvent(ObjectType):
+    class Meta:
+        interfaces = (EventNodeInterface,)
+
+    payload = Field(GenericScalar, required=True)
+
+
+EVENT_TYPES: dict[str, ObjectType] = {
+    "infrahub.node.added": NodeAddedEvent,
+    "infrahub.branch.created": BranchCreateEvent,
+    "infrahub.branch.rebased": BranchRebaseEvent,
+    "infrahub.branch.deleted": BranchDeleteEvent,
```

Comment:
Do we have anything that links these keys to the models? I wonder if we should get rid of the `infrahub` component here and say that any event would start with `event.xyz.xyz`?

Reply:
Not yet but I'm planning to refactor the events to have the name of the event more explicitly defined.
Regarding the prefix for the events, happy to consider another format